src/commands/tags/TagFileSystem.ts

```diff
@@ -105,10 +132,11 @@ export class TagFileSystem extends AzExtTreeFileSystem<ResourceGroupTreeItem | A
         return `// ${comment}${os.EOL}${JSON.stringify(tags, undefined, 4)}`;
     }
 
-    private async getTagsFromNode(node: ResourceGroupTreeItem | AppResourceTreeItem): Promise<Tags | undefined> {
-        if (node instanceof ResourceGroupTreeItem) {
-            return (await node.getData())?.tags;
-        }
-        return node.data.tags;
+    private async getTagsFromNode(node: ITagsModel): Promise<Tags | undefined> {
+        return await node.getTags();
+    }
+
+    private getDisplayName(node: ITagsModel): string {
+        return `${node.displayType} "${node.displayName}"`;
```

Comment:
A little bit confusing that there is a displayName property that is different from getDisplayName. Maybe something like getDetailedName or getFullName? getDisplayTypeName?